File: sum-of-positive.js
/*
You get an array of numbers, return the sum of all of the positives ones.

Example [1,-4,7,12] => 1 + 7 + 12 = 20

Note: if there is nothing to sum, the sum is default to 0.
*/

function positiveSum(arr) {
    //Declare variable to hold the sum value
    let sum = 0;
    //Loop through the array
    arr.forEach(element => {
      //If element is positive, add it to the sum value
      if (element >= 0){
        sum += element
      }
    })
    //Return the sum after looping through the array of numbers
    return sum
  }

File: python/8ky/sum-of-positive.py
'''
You get an array of numbers, return the sum of all of the positives ones.

Example [1,-4,7,12] => 1 + 7 + 12 = 20

Note: if there is nothing to sum, the sum is default to 0.
'''

def positive_sum(arr):
    sum = 0
    for value in arr:
        if value > 0:
            sum += value
    return sum if sum > 0 else 0

def positive_sum(arr):
    return sum(value for value in arr if value > 0)